share/spack/templates/container/centos_stream.dockerfile

```
@@ -19,10 +19,10 @@ RUN dnf update -y \
         iproute \
         make \
         patch \
-        python38 \
-        python38-pip \
-        python38-setuptools \
+        python3.11 \
+        python3.11-setuptools \
         unzip \
+ && python3.11 -m ensurepip \
```

```suggestion
 && python3 -m ensurepip \
```

Looking at `python3 --version` it seems like python3.11 get's aliased to python3 during installation, so simplifying here might help us reduce the number of lines we need to update on new centos stream releases.

---

Wouldn't that get out of sync as soon as:
```
dnf --enablerepo epel groupinstall -y "Development Tools"
```
bumps the python installation ?

---

Well, mystery solved:
```
# dnf repoquery -l python3.8-pip
Last metadata expiration check: 0:14:51 ago on Mon Jul 10 21:17:08 2023.
```

---

Using `pip3.11` explicitly, as we discussed yesterday in PM - see [595d842](https://github.com/spack/spack/pull/38824/commits/595d842eb77ed26809f1f90955d02b8bb519862b)